core/src/main/java/org/testcontainers/utility/ResourceReaper.java

```diff
@@ -145,22 +142,34 @@ public void removeNetworks(String identifier) {
     }
 
     private void removeNetwork(String networkName) {
-        List<Network> networks;
         try {
-            networks = dockerClient.listNetworksCmd().withNameFilter(networkName).exec();
-        } catch (DockerException e) {
-            LOGGER.trace("Error encountered when looking up network for removal (name: {}) - it may not have been removed", networkName);
-            return;
-        }
+            try {
+                // First try to remove by name
+                dockerClient.removeNetworkCmd(networkName).exec();
+            } catch (Exception e) {
+                LOGGER.trace("Error encountered removing network by name ({}) - it may not have been removed", networkName);
```

Comment:
I'm reusing across builds, on same machine. Quite similar to docker compose. I can code it with the client my self, but I think it's harmless  to leave the opening to set the name to who ever needs it. I'm ok with any direction you take.